subsystems/shootingsubsystem.py

```diff
@@ -0,0 +1,166 @@
+from commands2 import SubsystemBase
+from wpilib import SmartDashboard, RobotBase
+
+from wpimath.geometry import Rotation2d
+from util.angleoptimize import optimizeAngle
+from util.convenientmath import map_range
+from util.helpfulIO import Falcon
+
+import constants
+
+
+class ShootingSubsystem(SubsystemBase):
+    def __init__(self) -> None:
+        SubsystemBase.__init__(self)
+        # actuators
+        self.turretMotor = Falcon(
+            constants.kTurretMotorName,
+            constants.kTurretMotorId,
+            constants.kSimTurretMotorPort,
+            constants.kTurretPGain,
+            constants.kTurretIGain,
+            constants.kTurretDGain,
+            constants.kTurretPIDSlot,
+            constants.kTurretMotorInverted,
+        )
+        self.shootingMotor = Falcon(
+            constants.kShootingMotorName,
+            constants.kShootingMotorId,
+            constants.kSimShootingMotorPort,
+            constants.kShootingPGain,
+            constants.kShootingIGain,
+            constants.kShootingDGain,
+            constants.kShootingPIDSlot,
+            constants.kShootingMotorInverted,
+        )
+        self.hoodMotor = Falcon(
+            constants.kHoodMotorName,
+            constants.kHoodMotorId,
+            constants.kSimHoodMotorPort,
+            constants.kHoodPGain,
+            constants.kHoodIGain,
+            constants.kHoodDGain,
+            constants.kHoodPIDSlot,
+        )
+
+        self.initializationMinimum = 0
+        self.initializationMaximum = 0
+
+        self.targetTurretAngle = Rotation2d()
+        self.targetHoodAngle = Rotation2d()
+        self.targetWheelSpeed = 0
+
+    def setAsStartingPosition(self) -> None:
+        self.hoodMotor.setCurrentEncoderPulseCount(constants.kHoodStartingAngle)
+        if (
+            RobotBase.isReal()
+        ):  # only possible to calibrate the real robot, sim is perfect by default
+            turretRealPosition = map_range(
+                self.turretMotor.getPosition(),
+                self.initializationMinimum,
+                self.initializationMaximum,
+                constants.kTurretMinimum.radians()
+                * constants.kTalonEncoderPulsesPerRadian,
+                constants.kTurretMaximum.radians()
+                * constants.kTalonEncoderPulsesPerRadian,
+            )
+            self.turretMotor.setCurrentEncoderPulseCount(turretRealPosition)
+            self.turretMotor.setBrakeMode()
```

Comment:
This causes it to do that on reset too, which I don't think is intended

Reply:
why? I would think that resetting in general should do the same thing ever time and configure the same set of states

Reply:
Is that how the reset works? My understanding was that it's similar to the swerve drive reset in that we do it from a distance when it's screwed up. Is this to be used when doing the startup sequence (moving by hand)?

Reply:
Nvm, I think I misread and thought it was setting coast, which is isn't. Addressing that though, where are we setting coast?

Reply:
coast is default state from when we reset the entire falcon in general

Reply:
Define "reset the entire falcon", I know the steering motors on swerve are always on brake mode so what exactly do you mean?

Either way that sounds fine, maybe a good idea to have it be on coast when disabled tho if we can, that way we can recalibrate after if we have to

Reply:
when you do a configFactoryDefault, the factory default is coast mode

Reply:
also once the robot is enabled and reset there should be no reason it would ever be reset again

Reply:
As long as we don't manage to screw up the homing somehow, sure. When do we use factoryDefault? Usually we do it manually I thought

Reply:
factory default is set when the Falcon is initialized